src/api/utils/note-header.guard.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+
+import { ConsoleLoggerService } from '../../logger/console-logger.service';
+import { Note } from '../../notes/note.entity';
+import { NotesService } from '../../notes/notes.service';
+
+/**
+ * This guard take the note's id in header and find the Note in database.
+ */
+@Injectable()
+export class NoteHeaderGuard implements CanActivate {
```

Comment:
This should probably be an Interceptor instead of a Guard. Guards decide if a request is allowed to proceed, but this guard always returns `true` after manipulating the request object.

You can take a look at this interceptor we already implemented: https://github.com/hedgedoc/hedgedoc/blob/develop/src/api/utils/get-note.interceptor.ts

Reply:
Sure. I will change it to Interceptor